sandboxes/staticanalysis/staticanalyze.go

```diff
@@ -0,0 +1,235 @@
+		if err != nil {
+			return err
+		}
+		if f.Type().IsRegular() {
+			pathInArchive := strings.TrimPrefix(path, extractedPackageDir+string(os.PathSeparator))
+			fmt.Printf("Processing file %s\n", pathInArchive)
+			rawData, err := obfuscation.CollectData(jsParserPath, path, "", false)
+			if err != nil {
+				log.Error("Error parsing file", "filename", pathInArchive, "error", err)
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else if rawData == nil {
+				// syntax error - could not parse file
+				result.ExcludedFiles = append(result.ExcludedFiles, pathInArchive)
+			} else {
+				// rawData != nil, err == nil
+				result.FileRawData[f.Name()] = *rawData
+				signals := obfuscation.ComputeSignals(*rawData)
+				result.FileSignals[f.Name()] = obfuscation.RemoveNaNs(signals)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error while walking package files: %v", err)
+	}
+
+	return result, nil
+}
+
+func cleanupTmpDir(dir string) {
+	err := os.RemoveAll(dir)
+	if err != nil {
+		fmt.Printf("Failed to remove temp work dir: %v\n", err)
+	}
+}
+
+func makeWorkDirs() (tmpDir, archiveDir, extractDir string, err error) {
+	if tmpDir, err = os.MkdirTemp("", "package-analysis-staticanalyze"); err != nil {
+		return "", "", "", err
+	}
+
+	if archiveDir, err = os.MkdirTemp(tmpDir, "archive"); err != nil {
+		cleanupTmpDir(tmpDir)
+		return "", "", "", err
+	}
+
+	if extractDir, err = os.MkdirTemp(tmpDir, "extracted"); err != nil {
+		cleanupTmpDir(tmpDir)
+		return "", "", "", err
+	}
+
+	return
+}
+
+func mainReturningError() (err error) {
+	log.Initalize(os.Getenv("LOGGER_ENV"))
+	analyses.InitFlag()
+	flag.Parse()
+
+	if len(os.Args) == 1 || *help == true {
+		flag.Usage()
+		println()
+		printAnalyses()
+		return
+	}
+
+	if *ecosystem == "" || *packageName == "" {
+		flag.Usage()
+		return fmt.Errorf("ecosystem and package are required arguments")
+	}
+
+	manager := pkgecosystem.Manager(pkgecosystem.Ecosystem(*ecosystem))
+	if manager == nil {
+		return fmt.Errorf("unsupported pkg manager for ecosystem %s", *ecosystem)
+	}
+
+	pkg, err := worker.ResolvePkg(manager, *packageName, *version, *localFile)
+
+	if err != nil {
+		return fmt.Errorf("package error: %v", err)
+	}
+
+	uniqueAnalyses := utils.RemoveDuplicates(analyses.Values)
+	analysisTasks, err := checkAnalyses(uniqueAnalyses)
+
+	if err != nil {
+		printAnalyses()
+		return fmt.Errorf("%v\n", err)
+	}
+
+	log.Info("Static analysis launched",
+		log.Label("package", *packageName),
+		log.Label("version", *version),
+		log.Label("local_path", *localFile),
+		log.Label("analyses", strings.Join(uniqueAnalyses, ",")))
+
+	tmpDir, archiveDir, extractDir, err := makeWorkDirs()
+
+	defer cleanupTmpDir(tmpDir)
+
+	var archivePath string
+
+	if *localFile != "" {
+		// ensure we can access archive file to the archive dir
+		archivePath = archiveDir + string(os.PathSeparator) + *localFile
+		err = os.Link(*localFile, archivePath)
```

Comment:
Do we need to do this link trickery at all? Or can we just pass the original path down?

Reply:
Good point